pkg/algorithm/constants.go

```diff
@@ -32,9 +32,29 @@ const (
 	maxGuaranteedPriority = CellPriority(api.MaxGuaranteedPriority)
 	minGuaranteedPriority = CellPriority(api.MinGuaranteedPriority)
 	opportunisticPriority = CellPriority(api.OpportunisticPriority)
-	freePriority          = CellPriority(-2)
+	freePriority          = opportunisticPriority - 1
 
 	// lowest and highest levels in a cell chain
-	lowestLevel  = CellLevel(1)
-	highestLevel = CellLevel(math.MaxInt32)
+	lowestLevel  CellLevel = 1
+	highestLevel CellLevel = math.MaxInt32
+
+	// internal cell states
+
+	// no affinity group is using / acquiring or has acquired the cell
+	cellFree CellState = "Free"
+	// an affinity group is using this cell, and no other group is acquiring it
+	cellUsed CellState = "Used"
+	// an affinity group is using this cell, and another group is acquiring it
+	cellAcquiring CellState = "Acquiring"
+	// no affinity group is using this cell, and a group has acquired it
+	cellAcquired CellState = "Acquired"
+
+	// internal affinity group states
+
+	// the affinity group has been allocated cells and is allowed to run
+	groupAllocated affinityGroupState = "Allocated"
+	// the affinity group is waiting for the completion of preemption of other groups
```

Comment:
>waiting for the completion of preemption of other groups [](start = 26, length = 56)

it is not just waiting, it is proactively preempting others.
Just "the affinity group is preempting other groups to get free resource"?